Doc/whatsnew/3.7.rst

```diff
@@ -483,138 +782,458 @@ The :func:`~binascii.b2a_uu` function now accepts an optional *backtick*
+
+
+functools
+---------
+
+:func:`functools.singledispatch` now supports registering implementations
+using type annotations.
+(Contributed by Łukasz Langa in :issue:`32227`.)
+
+
+gc
+--
+
+The new :func:`gc.freeze`` function allows freezing all objects tracked
+by the garbage collector and exclude them from future collections.
+This can be used before a POSIX ``fork()`` call to make the GC copy-on-write
+friendly or to speed up collection.  The new :func:`gc.unfreeze` functions
+reverses this operation.  Additionally, :func:`gc.get_freeze_count` can
+be used to obtain the number of frozen objects.
+(Contributed by Li Zekun in :issue:`31558`.)
+
 
-:class:`distutils.core.setup` now warns if the ``classifiers``, ``keywords``
-and ``platforms`` fields are not specified as a list or a string.
-(Contributed by Berker Peksag in :issue:`19610`.)
+hmac
+----
+
+The :mod:`hmac` module now has an optimized one-shot :func:`~hmac.digest`
+function, which is up to three times faster than :func:`~hmac.HMAC`.
+(Contributed by Christian Heimes in :issue:`32433`.)
 
-The ``upload`` command no longer tries to change CR end-of-line characters
-to CRLF.  This fixes a corruption issue with sdists that ended with a byte
-equivalent to CR.
-(Contributed by Bo Bayles in :issue:`32304`.)
 
 http.client
 -----------
 
-Add configurable *blocksize* to :class:`~http.client.HTTPConnection` and
-:class:`~http.client.HTTPSConnection` for improved upload throughput.
+:class:`~http.client.HTTPConnection` and :class:`~http.client.HTTPSConnection`
+now support the new *blocksize* argument for improved upload throughput.
 (Contributed by Nir Soffer in :issue:`31945`.)
 
+
 http.server
 -----------
 
-:class:`~http.server.SimpleHTTPRequestHandler` supports the HTTP
-``If-Modified-Since`` header.  The server returns the 304 response status if the
-target file was not modified after the time specified in the header.
+:class:`~http.server.SimpleHTTPRequestHandler` now supports the HTTP
+``If-Modified-Since`` header.  The server returns the 304 response status if
+the target file was not modified after the time specified in the header.
 (Contributed by Pierre Quentel in :issue:`29654`.)
 
-Add the parameter *directory* to the :class:`~http.server.SimpleHTTPRequestHandler`
-and the ``--directory`` to the command line of the module :mod:`http.server`.
+:class:`~http.server.SimpleHTTPRequestHandler` accepts the new *directory*
+argument, in addition to the new ``--directory`` command line argument.
 With this parameter, the server serves the specified directory, by default it
 uses the current working directory.
 (Contributed by Stéphane Wirtel and Julien Palard in :issue:`28707`.)
 
-hmac
-----
+The new :class:`ThreadedHTTPServer <http.server.ThreadedHTTPServer>` class
+uses threads to handle requests using :class:`~socketserver.ThreadingMixin`.
+It is used when ``http.server`` is run with ``-m``.
+(Contributed by Julien Palard in :issue:`31639`.)
+
+
+idlelib and IDLE
```

Comment:
Since 3.4 IDLE changes are documented separately and should not be here. In particularly this change was backported to 3.7.